### PR TITLE
Update 2019-05-02-apollo-client-in-practice.md

### PR DESCRIPTION
**Pull request by mdvanes**

### CODE REVIEW

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 1]
```diff
@@ -96,8 +110,11 @@ to an array rows of cells
 # Next steps
 
 Apollo is excellent for merging data from multiple sources (in this case GraphQL, REST, local state and cache) and 
-functions as a "single source of truth" which should solve the state synchronization problems. These are some other 
-things you might want to take into account when working with Apollo Client.
+functions as a "single source of truth" which should solve the state synchronization problems. The local fields that
+Apollo uses in its local state manager can derive data, moving expensive operations from component render functions
+to resolvers in its application level cache. Although the issues mentioned in the introduction are now dealt with, we did 
+encounter plenty of other issues I may dive into later. However, these are some things that you might want to 
+take into account when working with Apollo Client.
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: `.` should be a `:` (I'm fine with the following listing not being formatted as a list.)

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 2]
```diff
@@ -114,7 +131,6 @@ encourages reactive behavior like RxJS (which can also be used as a [state store
 However, it seems that Apollo offers much less fine-grained control over the observables than what RxJS provides. And considering observables, Apollo Client
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: Comment about line 127: `willl` -> `will`

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 3]
```diff
@@ -7,14 +7,25 @@ tags:
...
+optimize performance on the side of network requests. 
 
 Looking into the performance tab of Chrome dev tools lead me to believe the performance problems were caused by 
 computations in the bottom components (e.g. formatting in cells). Because there are so many and they are 
 re-rendered quite often, this approach is quite intensive on resources.
 
-Besides that, the application had obvious state synchronization problems. When moving between views it was not  maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally. 
+Besides that, the application had obvious state synchronization problems. When moving between views it was not 
+maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it 
+between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally.
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: `it is not used consistently` -> `it was not used consistently`?

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 4]
```diff
@@ -25,14 +36,17 @@ Since both issues are caused by (a lack of) architecture, we redesigned the stru
...
 
-# Local state
+# Fixing application state with Apollo Local State
 
 When restructuring the application, Apollo Client was updated to 2.5. This version has a built-in [local state manager](https://www.apollographql.com/docs/react/essentials/local-state)
 (formerly *apollo-link-state*). and it supports REST calls with the [apollo-link-rest](https://www.apollographql.com/docs/link/links/rest) plugin. 
 The [apollo-boost](https://github.com/apollographql/apollo-client/tree/master/packages/apollo-boost) package contains the 
-client and several useful plugins. Adopting these means that both Redux and Axios can be removed and Apollo will be used as a single source of truth.
+client and several useful plugins. Adopting these means that both Redux and Axios can be removed and Apollo will be 
+used as a single source of truth. And if there is a single store for the data, there is no need for synchronization and
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: `And` at start of sentence can be removed.

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 5]
```diff
@@ -25,14 +36,17 @@ Since both issues are caused by (a lack of) architecture, we redesigned the stru
...
+with that one of the issues is solved. 
 
-Instead of using `client.query()` directly, components are split into a presentational component and enhanced with the 
+The way Apollo Client is used was also updated, to create a better separation of UI and data. Instead of using `client.query()`
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: `The way we used Apollo Client was also updated...`

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 6]
```diff
@@ -57,14 +71,14 @@ Apollo reactively updates when using `Query` as a container, basically like the
 the `variables` prop on the `Query` component is updated, it will automatically re-query. It use the cache if possible and
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: `It use the cache` -> `It uses the cache`

File: content/posts/2019-05-02-apollo-client-in-practice.md (hunk 3 above)
Comment (mdvanes) [reply content/posts/2019-05-02-apollo-client-in-practice.md]: I have the tendency to "forget" what tense I am writing in. I fixed it in some other spots.

File: content/posts/2019-05-02-apollo-client-in-practice.md (hunk 1 above)
Comment (mdvanes) [reply content/posts/2019-05-02-apollo-client-in-practice.md]: I wondered about this before submitting, but I modified it.

File: content/posts/2019-05-02-apollo-client-in-practice.md (hunk 3 above)
Comment (NRBPerdijk) [reply content/posts/2019-05-02-apollo-client-in-practice.md]: A very common problem.

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 7]
```diff
@@ -19,15 +19,15 @@ computations in the bottom components (e.g. formatting in cells). Because there
...
 Besides that, the application had obvious state synchronization problems. When moving between views it was not 
-maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it 
-between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally. 
+maintaining the same state of selected rows. Even though Redux was used to store application state and communicate it 
+between components, it was not used consistently. There were still plenty of React class components that stored some parts of the state locally. 
 
-To summarize, there are two issues that need to be solved:
+To summarize, there were two issues that needed to be solved:
 
 1. Reduce re-rendering to improve performance
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: Technically, these are two solutions. The issues would appear to be:

Poor performance due to needless re-rendering
Loss of state due to fractured application state

File: content/posts/2019-05-02-apollo-client-in-practice.md [hunk 8]
```diff
@@ -42,10 +42,10 @@ When restructuring the application, Apollo Client was updated to 2.5. This versi
 (formerly *apollo-link-state*). and it supports REST calls with the [apollo-link-rest](https://www.apollographql.com/docs/link/links/rest) plugin.
```
Comment (NRBPerdijk) [inline content/posts/2019-05-02-apollo-client-in-practice.md]: wrong fullstop :-)
